docs/infrastructure/vs-deps-branches.md

```diff
@@ -0,0 +1,132 @@
+# Why do we have -vs-deps branches?
+
+The Roslyn repo includes projects that reference Visual Studio APIs. These projects create a challenge because we have two tensions pulling us in opposite directions:
+
+1. We often want this to be able to reference the latest Visual Studio APIs, even if the APIs are not yet shipped. If the editor team creates a new API for us to implement or consume, they would
+   like us to consume it as soon as they have merged their feature work into the (internal) Visual Studio master branch; this means they can get API feedback and shake out any bugs as soon as possible.
+2. We want public contributors to be able to contribute to all of Roslyn, the Visual Studio layers included. Even if the contributor isn't _changing_ the code in the Visual Studio layer, deploying to Visual
+   Studio is the only way to actually _test_ any changes made to any of our layers in the middle in an interactive way. A contributor making a change to a code fix or refactoring can of course write unit
+   tests, but they can't just "try it locally" to experiment.
+
+To find a balance between these two tensions, we have two branches. Using our master branch as an example, we have two branches named 'master' and 'master-vs-deps':
```

Comment:
main-vs-deps is now gone :)
Not sure about "specificRelease"-vs-deps.

Reply:
We plan to avoid -vs-deps most of the time and recreate it temporarily if it is needed.